src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java

```diff
@@ -348,11 +348,13 @@ public static ArrayList parseHistoricalFile(String filename)
         } catch (InterruptedException e) {
           e.printStackTrace();
         }
+      } else {
+        // Check for line with greater than 0 non-whitespace length
+        if (line.trim().length() > 0) {
```

Comment:
the code checks for header here and sets the flag, but line is overwritten at the end of the while loop, and then headerEncoutered is checked again, but the line is different.

Reply:
That is correct. We are checking that the header line has already been seen (encountered), and if it has, only then do we parse any line for data.

Reply:
I missed the already part. It makes sense now.